visions/using-c++-from-swift.md

```diff
@@ -153,69 +169,82 @@ Or this API which fills in a pointer that has two levels of indirection:
 void VectorLike::begin(int **out) { *out = data(); }
 ```
 
-Or even this global function that projects one of it's parameters:
+Or even this global function that projects one of its parameters:
 ```
 int *begin(std::vector<int> *v) { return v->data(); }
 ```
 
-It may be convient for Swift to assume that all projects follow one, unique pattern: a method of an owned type that returns a pointer. However, that is certainly not the only way in which a projection can be created. This is but one of the many places where Swift will need to decide between expressiveness and safety. Allowing the above APIs to be imported would allow interop to be more usable by default. Taking the first example, most of the time, when a vector holds pointers, those pointers do not point to storage with a short lifetime. Making this API unavailable would ensure 100% safety on the pain usability in the 99% case, when this API is safe. The tradeoffs here are an open question for the Swift evolution process to eventually determine. In any case, it is essential that C++ interoperability makes certain assumptions about the APIs Swift imports. There will always be an edge case that cannot be covered or does not make sense to accomidate. Interop as a whole should not become unusable so that these edge cases can be accomidated, or worse yet, so that that neither safe nor unsafe APIs are available in Swift. So, C++ interoperability should make reasonable (not conservative) assumptions when importing APIs.
+Swift will need to decide how to handle projections, and more generally the use of view types, that it doesn't recognize how to make safe. This may come with difficult trade-offs between usefulness and safety. Consider the `projectsInternalStorage` API above, and pretend that we aren't able to recognize a pattern here --- we don't know that the pointers just depend on the `OwnedType` object. Often, uses of this kind of API can be made safe in practice: when there's a `std::vector` of pointers, there's probably some straightforward thing making those pointers valid. If Swift decides not to import this API just because it might be used unsafely, that could do serious damage to the usability of C++ interop. At the very least, it should be possible to wrap such an API in a safer Swift abstraction, which will be impossible if  it isn't imported at all.
```

Comment:
> If Swift decides not to import this API just because it might be used unsafely, that could do serious damage to the usability of C++ interop.

I think the way I said this was a bit stronger and clearer. I think I'm going to bring that part back.